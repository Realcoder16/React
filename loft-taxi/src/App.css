.App-form-text{

display: flex;
justify-content: flex-start;


}

body{
  background: black;
  color: white;
  
}




.App-header{
  background: white;
  height: 100px;
  display: flex;
  justify-content: space-between;
  align-items: center;

}

.App-list{
 
  margin-right: 205px;
  display: flex;
 
  list-style:none;

}

.App-list li{
  margin-left: 20px;
}


.map-wrapper{
  position: relative;
  width: 100vw;
  height: 100vh;
}

.map {

  position: absolute;
  top: 0;
  right: 0;
  height: 0;
  left: 0;
  width: 100%;
  height: 100%;
}

.taxi{
  position: absolute;
  top: 120px;
  right: 0;
  height: 0;
  left: 0;
}
