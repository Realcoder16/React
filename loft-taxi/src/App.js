import React from 'react';
import PropTypes from 'prop-types'
import './App.css';
import { Map } from "./Map"
import { ProfileWithAuth } from "./Profile"
import { LoginWithAuth } from "./Login"
import { Registration } from "./Registration"
import { Header } from "./Header"
import {withAuth} from './AuthContext'

  const PAGES = {
  Map: (props) => <Map {... props}/>,
  Profile: (props) =>  <ProfileWithAuth {... props}/>,
  Login: (props) => <LoginWithAuth {... props}/>,
  Registration: (props) =>  <Registration {... props} />,
 };



class App extends React.Component {

  state = { currentPage: "Login" }

  navigateTo = (page) => {
    if(this.props.isLoggedIn)
    {
      this.setState({ currentPage: page });
  
}else
{
  this.setState({currentPage: 'Login'});
}
    
  };
  render() { 

    return (
      <>
     <Header navigateTo={this.navigateTo}/>
               <main>
          <section>
          {PAGES[this.state.currentPage]({ navigate: this.navigateTo })}
          </section>
        </main>
      </>

    );
  }

} 

App.propTypes = {
  isLoggedIn: PropTypes.bool
};

export default withAuth(App);
