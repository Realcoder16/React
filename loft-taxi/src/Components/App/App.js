import React from "react";
import PropTypes from "prop-types";
import "./App.css";
import Map from "../Map/Map";
import Profile from "../Profile/Profile";
import Login from "../Login/Login";
import Registration from "../../Registration";
import { Header } from "../../Header";

import { Switch, Route } from "react-router-dom";
import { PrivateRoute } from "../../PrivateRoute";
import { createGlobalStyle } from "styled-components";

const GlobalStyle = createGlobalStyle`
body{background-color: black;}`;

class App extends React.Component {
  render() {
    return (
      <>
        <GlobalStyle />
        <Header className={this.props.className} />
        <main>
          <section>
            <Switch>
              <Route exact path="/" component={Login} />
              <Route exact path="/registration" component={Registration} />
              <PrivateRoute path="/map" component={Map} />
              <PrivateRoute path="/profile" component={Profile} />
            </Switch>
          </section>
        </main>
      </>
    );
  }
}

App.propTypes = {
  navigateTo: PropTypes.func,
};

export default App;
