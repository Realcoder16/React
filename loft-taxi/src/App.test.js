import React from 'react';

import ReactDOM from 'react-dom';
import { render, screen } from '@testing-library/react';
import App from './App';
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
it('renders shallow', ()=>{

  const wrapper = shallow(<App/>);
  expect(wrapper.contains(<Header/>)).toEqual(true);
});

it('renders correctly', ()=>{
  const tree  = renderer.create(<App/>).toJSON();
  expect(tree).toMatchSnapshot()
});