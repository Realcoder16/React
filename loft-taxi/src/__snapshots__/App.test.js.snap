// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <header
    className="App-header"
  >
    <div
      className="logo"
    >
      <img
        alt="logo"
        src="./img/лого1.gif"
      />
    </div>
    <nav>
      <ul
        className="App-list"
      >
        <li>
          <button
            onClick={[Function]}
          >
            Карта
          </button>
        </li>
        <li>
          <button
            onClick={[Function]}
          >
            Профиль
          </button>
        </li>
        <li>
          <button
            onClick={[Function]}
          >
            Выйти
          </button>
        </li>
      </ul>
    </nav>
  </header>,
  <main>
    <section>
      <form>
        <h2
          className="App-form-title"
        >
          Войти
        </h2>
        <div
          className="App-form-text"
        >
          <div>
            Новый пользователь?
          </div>
          <div
            onClick={[Function]}
          >
            Зарегистрируйся
          </div>
        </div>
        <label
          htmlFor="email"
        >
          Имя пользователя*
        </label>
        <input
          id="email"
          name="email"
          size="email"
          type="email"
        />
        <label
          htmlFor="password"
        >
          Пароль*
        </label>
        <input
          id="password"
          name="password"
          size="28"
          type="password"
        />
        <label
          htmlFor="submit"
        />
        <input
          form="data"
          onClick={[Function]}
          type="submit"
          value="Войти"
        />
      </form>
    </section>
  </main>,
]
`;
